src/test/ui/macros/macro-in-expression-context.stderr

```diff
@@ -0,0 +1,15 @@
+error: macro expansion ignores token `assert_eq` and any following
+  --> $DIR/macro-in-expression-context.rs:6:9
+   |
+LL |         assert_eq!("B", "B");
+   |         ^^^^^^^^^
+...
+LL |     foo!()
+   |     ------- help: you might be missing a semicolon here: `;`
+   |     |
+   |     caused by the macro expansion here
+   |
+   = note: the usage of `foo!` is likely invalid in expression context
```

Comment:
@dtolnay can you take a look and verify wether this suggestion is reasonable (it won't be wrong more often than not)?

Reply:
This is one case where it'd be the wrong approach:

```
fn main() {
    macro_rules! a {
        ($e:expr) => { $e; 2}
    }
    a!(true);
}
```
```
error: macro expansion ignores token `2` and any following
 --> file.rs:3:28
  |
3 |         ($e:expr) => { $e; 2}
  |                            ^
4 |     }
5 |     a!(true)
  |     --------- help: you might be missing a semicolon here: `;`
  |     |
  |     caused by the macro expansion here
  |
  = note: the usage of `a!` is likely invalid in expression context
```

but the suggestion actually gets the code to compile...